Src/Test.UnitTests.Sarif.PatternMatcher.Function/HttpAnalyzeFunctionTests.cs

```diff
@@ -106,5 +92,54 @@ public async Task Function_HttpAnalyze_WithEmptyFileName_Should_Return_OK()
             resultObject.Should().NotBeNull();
             resultObject.StatusCode.Should().Be((int)HttpStatusCode.OK);
         }
+
+        [Fact]
+        public async Task Function_HttpAnalyze_WithFileContent_Should_Return_DifferentResponses()
+        {
+            const string patTextFile = "SEC101_005.SlackTokens.py";
+            string content = TestHelper.GetTestResourceContent(patTextFile);
+            string[] lines = content.Split(Environment.NewLine);
+
+            IActionResult result = await HttpAnalyzeFunction.Analyze(
+                request: TestHelper.MockAnalyzeFunctionRequest(patTextFile, lines[0]),
+                log: logger,
+                context: TestHelper.ContextSetup());
+            ValidateResult(lines[0], result, runCount: 1, resultCount: 1, FailureLevel.Warning);
+
+            result = await HttpAnalyzeFunction.Analyze(
+               request: TestHelper.MockAnalyzeFunctionRequest(patTextFile, lines[1]),
+               log: logger,
+               context: TestHelper.ContextSetup());
+            ValidateResult(lines[1], result, runCount: 1, resultCount: 0, FailureLevel.None);
+        }
+
+        private static void ValidateResult(string text, IActionResult result, int runCount, int resultCount, FailureLevel failureLevel = FailureLevel.None, bool ignoreRegionContent = false)
+        {
+            var resultObject = (OkObjectResult)result;
+            resultObject.Should().NotBeNull();
+            resultObject.StatusCode.Should().Be((int)HttpStatusCode.OK);
+            resultObject.Value.Should().NotBeNull();
+
+            var sarifLog = resultObject.Value as SarifLog;
+            sarifLog.Should().NotBeNull();
+            sarifLog.Runs.Count.Should().Be(runCount);
+
+            IList<Result> results = sarifLog.Runs[0].Results;
+
+            if (resultCount == 0)
```

Comment:
If we're checking a specific file, can't we say for certain whether or not it will have results?